nrf9160dk-keys/src/main.c

```diff
@@ -0,0 +1,203 @@
+        else printk("Success!\n");
+    }
+    return 0;
+}
+
+int delete_credentials_from_modem(bool id_in_modem, bool psk_in_modem)
+{
+    int err;
+
+    if ((id_in_modem | psk_in_modem) == 0)
+    {
+        /* This should never happen */
+        printk("ERROR: No credentials available to erase from modem\n");
+        return -EINVAL;
+    }
+    printk("Attempting to delete credentials with sec_tag=1 from modem...\n");
+    if (id_in_modem)
+    {
+        printk("Deleting PSK-ID from modem...\n");
+        err = modem_key_mgmt_delete(1, MODEM_KEY_MGMT_CRED_TYPE_IDENTITY);
+        if (err) printk("Could not delete PSK-ID: %d", err);
+        else printk("Success!\n");
+    }
+    if (psk_in_modem)
+    {
+        printk("Deleting PSK from modem...\n");
+        err = modem_key_mgmt_delete(1, MODEM_KEY_MGMT_CRED_TYPE_PSK);
+        if (err) printk("Could not delete PSK: %d", err);
+        else printk("Success!\n");
+    }
+    return 0;
+}
+
+void show_prompt(bool in_modem, bool in_prj) {
+    printk("\nManage Modem PSK-ID/PSK Storage:\n");
+    printk("================================\n\n");
+    if ((in_modem | in_prj) == 0)
+    {
+        printk("No credentials found in modem or in Kconfig (ie: prj.conf\n");
+        printk("Nothing to do here!\n\n");
+        return;
+    }
+    if (in_prj)
+    {
+        printk("Press Button 1 to write Golioth credentials from Kconfig (ie: prj.conf) to modem\n");
+    }
+    if (in_modem)
+    {
+        printk("Press Button 2 to ERASE the PSK-ID and PSK credentials with sec_tag=1 from modem (DANGER!)\n");
+    }
+    printk("\n");
+}
+
+int check_credentials(bool *psk_id_in_modem, bool *psk_in_modem, bool *psk_id_in_prj, bool *psk_in_prj)
+{
+    int err;
+    static const char FOUND[] = "FOUND";
+    static const char ABSENT[] = "ABSENT";
+
+    printk("\nChecking credentials...\n");
+    printk("=======================\n");
+
+    /* Check for credentials in modem */
+    err = modem_key_mgmt_exists(1, MODEM_KEY_MGMT_CRED_TYPE_IDENTITY, psk_id_in_modem);
+    if (err) {
+        printk("Error reading PSK-ID from modem: %d\n", err);
+    }
+    else
+    {
+        printk("%10s: PSK-ID with sec_tag=1 stored in modem\n", *psk_id_in_modem ? FOUND : ABSENT);
+    }
+    err = modem_key_mgmt_exists(1, MODEM_KEY_MGMT_CRED_TYPE_PSK, psk_in_modem);
+    if (err) {
+        printk("Error reading PSK from modem: %d\n", err);
+    }
+    else
+    {
+        printk("%10s: PSK with sec_tag=1 stored in modem\n", *psk_in_modem ? FOUND : ABSENT);
+    }
+
+    /* Check for credentials in prj.conf */
+    #ifdef CONFIG_GOLIOTH_SYSTEM_CLIENT_PSK_ID
+    if (strlen(CONFIG_GOLIOTH_SYSTEM_CLIENT_PSK_ID) > 0) *psk_id_in_prj = 1;
+    #endif
+    printk("%10s: Golioth PSK-ID from Kconfig (ie: prj.conf)\n", *psk_id_in_prj ? FOUND : ABSENT);
+
+    #ifdef CONFIG_GOLIOTH_SYSTEM_CLIENT_PSK
+    if (strlen(CONFIG_GOLIOTH_SYSTEM_CLIENT_PSK) > 0) *psk_in_prj = 1;
```

Comment:
Nit: As a matter style, it's preferred to have `{}` on all if statements and avoid single-line if statements.

```c
if (strlen(CONFIG_GOLIOTH_SYSTEM_CLIENT_PSK) > 0) 
{
    *psk_in_prj = 1;
}
```

You could also write it like this:

```c
*psk_id_in_prj = (strlen(CONFIG_GOLIOTH_SYSTEM_CLIENT_PSK_ID) > 0);
```